src/index.ts

```diff
@@ -72,17 +72,15 @@ const isDevelopmentVersion = (version: string) => {
  * resinSemver.compare('Resin OS 1.16.0', 'Resin OS 1.16.0'); //--> 0
  */
 export const compare = memoize((versionA: string | null, versionB: string | null): number => {
-	if (versionA === null && versionB === null) {
-		return 0;
+	if (versionA === null) {
+		return versionB === null ? 0 : -1;
 	}
-	if (versionA === null && versionB !== null) {
-		return -1;
+	if (versionB === null) {
+		return versionA === null ? 0 : 1;
```

Comment:
It doesn't cause any problems, but you don't actually need this 2nd inner check - if you've got to this line, you already know that `versionA !== null`